include/mbgl/actor/scheduler.hpp

```diff
@@ -50,6 +50,18 @@ class Scheduler {
     // first invoked, the call is ignored.
     std::function<void()> bindOnce(std::function<void()>);
 
+    // Enqueues the given |task| for execution into this scheduler's task queue and
+    // then enqueues the |reply| with the captured task result to the current
+    // task queue.
+    //
+    // The |Function| return type must be compatible with the |Reply| argument type.
+    // Note: the task result is copied and passed by value.
+    template <typename Function, typename Reply>
+    void scheduleAndReplyValue(const Function& task, const Reply& reply) {
+        assert(GetCurrent());
```

Comment:
Reply should be invocable? With C++17 have nice tools to add static check if reply is invocable with result of a task.